hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py

```diff
@@ -87,21 +88,22 @@ def check_employee_attendance_state(self):
                 )
             )
 
-    @api.model
-    def create(self, vals):
-        res = super().create(vals)
-        attendances = self.env["hr.attendance"].search(
-            [
-                ("employee_id", "=", res.employee_id.id),
-                ("sheet_id", "=", False),
-                ("check_in", ">=", res.date_start),
-                ("check_in", "<=", res.date_end),
-                "|",
-                ("check_out", "=", False),
-                "&",
-                ("check_out", ">=", res.date_start),
-                ("check_out", "<=", res.date_end),
-            ]
-        )
-        attendances._compute_sheet_id()
-        return res
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        for res in records:
+            attendances = self.env["hr.attendance"].search(
+                [
+                    ("employee_id", "=", res.employee_id.id),
+                    ("sheet_id", "=", False),
+                    ("check_in", ">=", res.date_start),
+                    ("check_in", "<=", res.date_end),
+                    "|",
+                    ("check_out", "=", False),
+                    "&",
+                    ("check_out", ">=", res.date_start),
+                    ("check_out", "<=", res.date_end),
+                ]
+            )
+            attendances._compute_sheet_id()
```

Comment:
just small comment but maybe you could initialize variable attendances = self.env["hr.attendance"], then on each iteration do attendances |= self.env["hr.attendance"].search...., and then only call the _compute_sheet_id method once outside the loop